### DISCUSSION TITLE
Questions about "fparam"

### DISCUSSION THREAD

**Discussion by Ymgraal** [deepmodeling/deepmd-kit#2341] [Q&A]
Dear all,

does anyone know if it is possible to train a model with atomic scalars for only one atom type ?

Let me explain.

We’re trying to fit the quadrupolar coupling of different types of nuclei for solid nmr study. This parameter is an atomic scalar and we would like to use one model per atom type. 

In that optic, we thought about using a fparam.npy with "numb_fparam": = the number of atoms in the atom type.

However we have two issues with this methodology: 

The first is that, because we want a scalar, the fitting net type is set to “energy” which does not permit the use of the “sel_type” keyword and thus we do not know how to specify which atom we are working on. So in that context how could we tell to the model what is the type we're interested in ?

The second one is that we do not know how to configure the loss, that is “type”: ener, because we can’t find a start/limit_pref that corresponds to the fparam.npy. So how can we configure the weight of the fparam in the loss function ?

thank you in advance for your help.

**Comment by wanghan-iapcm** [comment]
Is the atomic scalar the desired output of your model? Do you have the labels for the scalar?

**Reply by Ymgraal** [reply]
Thanks for your answer,
Yes the atomic scalar are the desired outputs and yes we have the labels. These are the atomic scalars i.e the quadrupolar coupling.

**Reply by wanghan-iapcm** [reply]
The fparam is the input of the model. 
You may fix a scalar output by the `atom_ener`, i.e. provide the quadrupolar coupling as atomic energy labels and train the standard DP model to predict the correct "atomic energy", then you will have a model for the quadrupolar coupling.

Please see the [doc](https://docs.deepmodeling.com/projects/deepmd/en/master/data/system.html?highlight=atom_ener#system)

**Reply by Ymgraal** [reply]
Thank you again for your answer.
This is what we tried to do but we want to train the model with the quadrupolar coupling of only one type of atom. 
The problem we faced is that atomic energy has to be defined for all atoms. In our case we have the label i.e quadrupolar coupling of 40 atoms but the total number of atoms is 96, this is the error message we have while trying to train the model:

ValueError: cannot reshape array of size 8000 into shape (200,96,newaxis). This error may occur when your label mismatch it's name, i.e. you might store global tensor in `atomic_tensor.npy` or atomic tensor in `tensor.npy`.

**Reply by wanghan-iapcm** [reply]
You may hack the code by assigning the undesired atom types with any  constant value of `atom_ener`.  
Different atom types use different fitting net, thus it would not affect the training of interested atom type.

**Comment by njzjz** [comment]
The neural network parameters are atom type-wise by default. Thus, it's not necessary to train multiple models.
In other words, fitting type B will not affect type A, so you can give any values for type B if only type A is the target.
